docroot/profiles/custom/sitenow/sitenow.install

```diff
@@ -1950,3 +1951,116 @@ function sitenow_update_9054() {
   $config_storage = \Drupal::service('config.storage');
   $config_storage->write('acquia_connector.settings', $source->read('acquia_connector.settings'));
 }
+
+/**
+ * Remove all Large, Medium, and Small image view modes and replace them with relevant alternatives.
+ */
+function sitenow_update_9055() {
+  $entity_types = [
+    'block_content',
+    'media',
+    'node',
+    'paragraph',
+  ];
+
+  $media_view_mode_map = [
+    'small' => 'small__no_crop',
+    'medium' => 'small__no_crop',
+    'large' => 'medium__no_crop',
+  ];
+  $media_view_mode_map_keys = array_keys($media_view_mode_map);
+
+  foreach ($entity_types as $entity_type) {
+    $bundles = \Drupal::service('entity_type.bundle.info')
+      ->getBundleInfo($entity_type);
+
+    // @todo Handle case where no bundles exist?
+    foreach ($bundles as $bundle => $bundle_label) {
+      /** @var \Drupal\Core\Entity\EntityFieldManagerInterface $entityFieldManager */
+      $entityFieldManager = \Drupal::service('entity_field.manager');
+      $fields = $entityFieldManager->getFieldDefinitions($entity_type, $bundle);
+      $field_check = [];
+      /**
+       * @var string $field_name
+       * @var \Drupal\field\Entity\FieldConfig $definition
+       */
+      foreach ($fields as $field_name => $definition) {
+        if (in_array($definition->getType(), [
+          'text_long',
+          'text_with_summary',
+        ])) {
+          $field_check[$field_name] = $definition;
+        }
+      }
+
+      // Only need to run the following if we found fields to check against.
+      if (!empty($field_check)) {
+
+        // Retrieve all entities of this bundle type;.
+        // Media is listed as 'bundle,' but other entity types
+        // are listed as 'type'.
+        $type = $entity_type == 'media' ? 'bundle' : 'type';
+        $entities = \Drupal::entityTypeManager()->getStorage($entity_type)
+          ->loadByProperties([$type => $bundle]);
+
+        // Loop through the loaded entities.
+        foreach ($entities as $entity) {
+          $needs_save = FALSE;
+
+          // Loop through the fields we defined before.
+          foreach ($field_check as $name => $definition) {
+            // If the field has a value (or values).
+            if (!$entity->get($name)->isEmpty()) {
+              $updated_values = [];
+              $field_values = $entity->{$name}->getValue();
+
+              foreach ($field_values as $field_value) {
+
+                // @todo Any special handling for text_with_summary fields?
```

Comment:
There shouldn't need any special handling for `text_with_summary`. The regular value is still in `['value']`, and if there does happen to be something in the `['summary']` (copy/pasting?), it won't be rendered, and there's a good chance it'd be truncated and broken anyway.

I think the main chance there'd be for something ending up in there would have been migrations, but we have special handling for those cases that _should_ have prevented that.
```suggestion
```

Reply:
On the flip side, it'd be possible to check the string format before processing, and only proceed with those that have a format that allows for embeds.

But those cases would just end up with an `$i` of `-1`, and processing would get skipped anyway, so 🤷